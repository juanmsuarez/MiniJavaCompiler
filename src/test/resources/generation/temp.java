class A {
    static void f() {
        System.println();
    }

    dynamic void g() {
        if (true) {
            int a;
            a = 1;
            System.printIln(a);
        }

        int x; x = 0;
        while (x < 2) {
            int a;
            a = 2;

            if (x < 1) {
                int y; y = 3;
                System.printIln(y);
            }

            System.printIln(a);
            x = x + 1;

        }

        int a, y; a = 4; y = 5;
        System.printIln(a);
        System.printIln(y);

        char c; c = 'a';
        System.printCln(c);
        System.println();
    }
}

class B extends A {
    static void f() {
        if (false) {
            int a, b, c;
            a = 1; b = 2; c = 3;
            System.printIln(a);
            System.printIln(b);
            System.printIln(c);
        }
        System.println();
    }

    dynamic void g() {
        boolean b; b = true;
        System.printBln(b);
        System.println();
    }
}

class Main {
    static void main() {
        int x, y;
        x = 1; y = 2;
        System.printIln(x);
        System.printIln(y);
        System.println();

        A a; a = new A();
        a.f();
        a.g();

        A b; b = new B();
        b.f();
        b.g();
    }
}
