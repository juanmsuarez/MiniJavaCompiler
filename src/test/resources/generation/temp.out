.CODE
PUSH HELPER_HEAP_INIT
CALL
PUSH METHOD_Main_main
CALL
HALT
HELPER_HEAP_INIT:
RET 0
HELPER_MALLOC:
LOADFP
LOADSP
STOREFP
LOADHL
DUP
PUSH 1
ADD
STORE 4
LOAD 3
ADD
STOREHL
STOREFP
RET 1
.DATA
VT_A: DW METHOD_A_g
.CODE
CONSTRUCTOR_A: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_A_f: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
FMEM 0
STOREFP
RET 0
.CODE
METHOD_A_g: LOADFP
LOADSP
STOREFP
RMEM 2
.CODE
PUSH 1
.CODE
STORE 0
.CODE
LOAD 0
.CODE
PUSH METHOD_System_printIln
CALL
.CODE
PUSH 97
.CODE
STORE -1
.CODE
LOAD -1
.CODE
PUSH METHOD_System_printCln
CALL
.CODE
FMEM 2
STOREFP
RET 1
.DATA
VT_B: DW METHOD_B_g
.CODE
CONSTRUCTOR_B: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_B_g: LOADFP
LOADSP
STOREFP
RMEM 1
.CODE
PUSH 1
.CODE
STORE 0
.CODE
LOAD 0
.CODE
PUSH METHOD_System_printBln
CALL
.CODE
FMEM 1
STOREFP
RET 1
.DATA
VT_Object: NOP
.CODE
CONSTRUCTOR_Object: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
FMEM 0
STOREFP
RET 1
.DATA
VT_Main: NOP
.CODE
CONSTRUCTOR_Main: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_Main_main: LOADFP
LOADSP
STOREFP
RMEM 4
.CODE
PUSH 1
.CODE
STORE 0
.CODE
PUSH 2
.CODE
STORE -1
.CODE
LOAD 0
.CODE
PUSH METHOD_System_printIln
CALL
.CODE
LOAD -1
.CODE
PUSH METHOD_System_printIln
CALL
.CODE
RMEM 1
PUSH 1
PUSH HELPER_MALLOC
CALL
DUP
PUSH VT_A
STOREREF 0
DUP
.CODE
PUSH CONSTRUCTOR_A
CALL
.CODE
STORE -2
.CODE
LOAD -2
.CODE
POP
.CODE
PUSH METHOD_A_f
CALL
.CODE
LOAD -2
.CODE
DUP
LOADREF 0
LOADREF 0
CALL
.CODE
RMEM 1
PUSH 1
PUSH HELPER_MALLOC
CALL
DUP
PUSH VT_B
STOREREF 0
DUP
.CODE
PUSH CONSTRUCTOR_B
CALL
.CODE
STORE -3
.CODE
LOAD -3
.CODE
DUP
LOADREF 0
LOADREF 0
CALL
.CODE
FMEM 4
STOREFP
RET 0
.DATA
VT_System: NOP
.CODE
CONSTRUCTOR_System: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_printC: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
CPRINT
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_printS: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
SPRINT
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_println: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
PRNLN
.CODE
FMEM 0
STOREFP
RET 0
.CODE
METHOD_System_printCln: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
CPRINT
PRNLN
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_printSln: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
SPRINT
PRNLN
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_read: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
READ
STORE 3
.CODE
FMEM 0
STOREFP
RET 0
.CODE
METHOD_System_printB: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
BPRINT
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_printIln: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
IPRINT
PRNLN
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_printI: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
IPRINT
.CODE
FMEM 0
STOREFP
RET 1
.CODE
METHOD_System_printBln: LOADFP
LOADSP
STOREFP
RMEM 0
.CODE
LOAD 3
BPRINT
PRNLN
.CODE
FMEM 0
STOREFP
RET 1
